UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.Collisions.cs

```diff
@@ -247,8 +251,75 @@ private void CheckTileCollisions( MatrixIntersection i )
 				continue;
 			}
 
+
+			if (i.Matrix1.MatrixMove.NetworkedMatrixMove.Safety)
```

Comment:
this whole thing is too big and complicated. Reduce nesting and make functions with self-explanatory names that say what each step do.

https://plugins.jetbrains.com/plugin/12024-cognitivecomplexity